performances/irobot-benchmark/src/irobot-benchmark.cpp

```diff
@@ -26,6 +26,68 @@
 
 using namespace std::chrono_literals;
 
+template <typename NodeT>
+void init_ros_system(int argc, char** argv,
+                    const irobot_benchmark::Options &options,
+                    const std::string &topology_json,
+                    pid_t &pid,
+                    const std::string &resources_output_path,
+                    const std::string &events_output_path,
+                    const std::string &latency_all_output_path,
+                    const std::string &latency_total_output_path)
+{
+    // Start resources logger
+    performance_test::ResourceUsageLogger ru_logger(resources_output_path);
+
+    ru_logger.start(std::chrono::milliseconds(options.resources_sampling_per_ms));
+
+    rclcpp::init(argc, argv);
+
+    performance_test::System<NodeT> ros2_system(static_cast<performance_test::ExecutorType>(options.executor));
```

Comment:
rather than `NodeT` being a template, can't we follow the same approach done for the executor and pass `performance_test::NodeType>(options.node)` ?

we should try to keep the benchmark application as clean as possible and just forward the options to system/factory classes